scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,423 @@
+            this.artifactName = IO_FABRIC_COORD
+            this.artifactVersion = IO_FABRIC_COORD_VERSION
+            this.addAllLicense(listOf(license))
+          }.build()
+        )
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val coordsList = listOf(IO_FABRIC_COORD)
+    setupBazelEnvironment(coordsList)
+
+    val exception = assertThrows(Exception::class) {
+      GenerateMavenDependenciesListRunner(
+        mockLicenseFetcher,
+        CommandExecutorImpl(600_000L)
+      ).main(
+        arrayOf(
+          "${tempFolder.root}",
+          "scripts/assets/maven_install.json",
+          "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+        )
+      )
+    }
+    assertThat(exception).hasMessageThat().contains(UNAVAILABLE_OR_INVALID_LICENSE_LINKS_FAILURE)
+  }
+
+  @Test
+  fun testDependenciesHaveMultipleLicense_licenseDetailsCompleted_scriptPasses() {
+    tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder()
+        .setUrl("https://www.apache.org/licenses/LICENSE-2.0.txt").build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+      this.localCopyLink = LocalCopyLink.newBuilder()
+        .setUrl("https://local-copy/bsd-license").build()
+    }.build()
+    val mavenDependencyList = MavenDependencyList.newBuilder().apply {
+      this.addAllMavenDependency(
+        listOf(
+          MavenDependency.newBuilder().apply {
+            this.artifactName = DATA_BINDING_COORD
+            this.artifactVersion = DATA_BINDING_COORD_VERSION
+            this.addAllLicense(listOf(license1))
+          }.build(),
+          MavenDependency.newBuilder().apply {
+            this.artifactName = GLIDE_ANNOTATIONS_COORD
+            this.artifactVersion = GLIDE_ANNOTATIONS_COORD_VERSION
+            this.addAllLicense(listOf(license1, license2))
+          }.build()
+        )
+      )
+    }.build()
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val coordsList = listOf(DATA_BINDING_COORD, GLIDE_ANNOTATIONS_COORD)
+    setupBazelEnvironment(coordsList)
+
+    GenerateMavenDependenciesListRunner(
+      mockLicenseFetcher,
+      CommandExecutorImpl(600_000L)
+    ).main(
+      arrayOf(
+        "${tempFolder.root}",
+        "scripts/assets/maven_install.json",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_MESSAGE)
+  }
+
+  @Test
+  fun testDependenciesHaveCompleteLicenseDetails_scriptPasses() {
```

Comment:
What exactly is the difference between this & the test above?

Reply:
Not much, in the above test case, all the dependencies have almost one license.